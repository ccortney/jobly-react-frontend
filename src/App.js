import {useState, useEffect} from "react";
import { useLocalStorage } from "./hooks";
import AppRoutes from "./AppRoutes";
import JoblyApi from "./api";
import { decodeToken } from 'react-jwt';
import UserContext from "./UserContext";
import './App.css';
import Navbar from "./Navbar";


function App() {

  const [token, setToken] = useLocalStorage("token", null);
  const [user, setUser] = useState(null);

  useEffect(() => {
    async function getUserFromToken() {
      JoblyApi.token = token;
      let payload = decodeToken(token)
      const user = await JoblyApi.getUser(payload.username)
      setUser(user)
    }
    if (token) {
      getUserFromToken()
    }
  }, [token])

  const login = async (data) => {
    let res = await JoblyApi.login(data);
    setToken(res)
  };

  const signup = async (data) => {
    let res = await JoblyApi.signup(data);
    setToken(res)
  }

  const editProfile = async (data) => {
    let res = await JoblyApi.editProfile(user.username, data);
    setUser(res);
  }

  const applyForJob = async (jobId) => {
    let res = await JoblyApi.applyForJob(user.username, jobId);
    setUser({...user, applications: [...user.applications, res]})
  }

  const logout = () => {
    JoblyApi.token = null;
    setToken(null)
    setUser(null)
  }

  return (
    <UserContext.Provider value={user}>
      <div className="App">
        <Navbar logout={logout} />
        <AppRoutes login={login} signup={signup} token={token} editProfile={editProfile} applyForJob={applyForJob}/>
      </div>
    </UserContext.Provider>

  );
}

export default App;
