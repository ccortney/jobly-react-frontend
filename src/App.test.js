import { render } from '@testing-library/react';
import App from './App';
import { MemoryRouter } from "react-router-dom";


test("should render without crashing", () => {
  render(<MemoryRouter><App/></MemoryRouter>)
});

test("should match snapshot", () => {
  const {asFragment} = render(<MemoryRouter><App /></MemoryRouter>);
  expect(asFragment()).toMatchSnapshot();
})


