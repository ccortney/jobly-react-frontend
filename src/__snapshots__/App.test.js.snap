// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular css-r6ewbb-MuiToolbar-root"
        >
          <a
            class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-11mmdhf-MuiTypography-root"
            href="/"
          >
            JOBLY
          </a>
          <div
            class="MuiBox-root css-19aihql"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1q39md6-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              <a
                href="/companies"
                style="text-decoration: none; color: white;"
              >
                companies
              </a>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1q39md6-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              <a
                href="/jobs"
                style="text-decoration: none; color: white;"
              >
                jobs
              </a>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiBox-root css-7ln23k"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-is4nal-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              <a
                href="/signup"
                style="text-decoration: none; color: white;"
              >
                signup
              </a>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-is4nal-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              <a
                href="/login"
                style="text-decoration: none; color: white;"
              >
                login
              </a>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </header>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column css-iw00ie-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 css-10fld7o-MuiTypography-root"
        >
          JOBLY
        </h2>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
        >
          All the jobs in one, convenient place.
        </h6>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          class="MuiBox-root css-0"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-bh137t-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <a
              href="/signup"
              style="text-decoration: none; color: white;"
            >
              Sign Up
            </a>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-axpn69-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <a
              href="/login"
              style="text-decoration: none; color: white;"
            >
              LOGIN
            </a>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
